### PR TITLE
feat(menu): manage deprecated menus (hosts and services monitoring)

### PR DESCRIPTION
**Pull request by kduret**
## Description

Allow to deprecate menus by adding a new column in database
A user can choose if he wants to show/hide deprecated menus

image

image

**Fixes** MON-5931

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x (master)

### CODE REVIEW

File: src/Centreon/Domain/Entity/Topology.php [hunk 1]
```diff
@@ -149,7 +155,13 @@ public function setTopologyId(int $topology_id): void
     public function getTopologyName(): ?string
     {
         // get translated menu entry
-        return _($this->topology_name);
+        $topologyName = _($this->topology_name);
+
+        if ($this->getIsDeprecated() === '1') {
+            $topologyName.= ' (' . _('deprecated') . ')';
```
Comment (kduret) [inline src/Centreon/Domain/Entity/Topology.php:161]:
```suggestion
            $topologyName .= ' (' . _('deprecated') . ')';
```

File: src/Centreon/Domain/Entity/Topology.php [hunk 2]
```diff
@@ -304,6 +316,22 @@ public function setTopologyShow(string $topology_show): void
...
+     */
+    public function getIsDeprecated(): ?string
+    {
+        return $this->is_deprecated;
+    }
+
+    /**
+     * @param string $isDeprecated
+     */
+    public function setIsDeprecated(string $isDeprecated): void
```
Comment (callapa) [inline src/Centreon/Domain/Entity/Topology.php:330]:
```suggestion
    public function setDeprecated(bool $isDeprecated): void
```

File: src/Centreon/Domain/Entity/Topology.php [hunk 3]
```diff
@@ -304,6 +316,22 @@ public function setTopologyShow(string $topology_show): void
         $this->topology_show = $topology_show;
     }
 
+    /**
+     * @return string
+     */
+    public function getIsDeprecated(): ?string
```
Comment (callapa) [inline src/Centreon/Domain/Entity/Topology.php:322]:
```suggestion
    public function IsDeprecated(): bool
```

File: src/Centreon/Domain/Entity/Topology.php [hunk 4]
```diff
@@ -99,6 +99,12 @@ class Topology
      */
     protected $topology_show;
 
+    /**
+     * @var string
+     * enum('0','1')
+     */
+    protected $is_deprecated;
```
Comment (callapa) [inline src/Centreon/Domain/Entity/Topology.php:106]: change the type to a boolean value and give it a default value (same as its database definition)

**Review by callapa — CHANGES_REQUESTED** [review]

File: src/Centreon/Domain/Entity/Topology.php (hunk 3 above)
Comment (kduret) [reply src/Centreon/Domain/Entity/Topology.php:322]: It is the legacy code. I can't change it cause it breaks entity creation

File: src/Centreon/Domain/Entity/Topology.php (hunk 2 above)
Comment (kduret) [reply src/Centreon/Domain/Entity/Topology.php:330]: It is the legacy code. I can't change it cause it breaks entity creation

**Review by callapa — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

File: src/Centreon/Domain/Entity/Topology.php [hunk 5]
```diff
@@ -149,7 +154,13 @@ public function setTopologyId(int $topology_id): void
     public function getTopologyName(): ?string
     {
         // get translated menu entry
-        return _($this->topology_name);
+        $topologyName = _($this->topology_name);
+
+        if ($this->getIsDeprecated() === '1') {
```
Comment (kduret) [inline src/Centreon/Domain/Entity/Topology.php:159]:
```suggestion
        if ($this->getIsDeprecated() === true) {
```

**Review by callapa — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

File: src/Centreon/Domain/Repository/TopologyRepository.php [hunk 6]
```diff
@@ -159,11 +159,19 @@ public function getTopologyList(CentreonUser $user): array
...
+            . 'topology_group, topology_order, topology_parent, is_react, readonly, topology_show, is_deprecated '
             . 'FROM ' . Topology::TABLE;
 
+        $whereClause = false;
         if (!$user->access->admin) {
             $query .= ' WHERE topology_page IN (' . $user->access->getTopologyString() . ')';
+            $whereClause = true;
+        }
+
+        //var_dump($user->doesShowDeprecatedPages());
```
Comment (kduret) [inline src/Centreon/Domain/Repository/TopologyRepository.php:171]:
```suggestion
```

File: www/class/centreonUser.class.php [hunk 7]
```diff
@@ -95,6 +96,7 @@ public function __construct($user = array())
         $this->default_page = $user["default_page"];
         $this->gmt = $user["contact_location"];
         $this->js_effects = $user["contact_js_effects"];
+        $this->showDeprecatedPages = boolval($user["show_deprecated_pages"]);
```
Comment (jeremyjaouen) [inline www/class/centreonUser.class.php:99]:
```suggestion
        $this->showDeprecatedPages = (bool) $user["show_deprecated_pages"];
```

**Review by jeremyjaouen — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**